openedx/adg/admin_dashboard/applications/helpers.py

```diff
@@ -224,3 +233,52 @@ def get_application_review_url(application_id, query_params_string):
         f'{application_review_url}?{query_params_string}' if query_params_string else application_review_url
     )
     return application_review_url
+
+
+def get_valid_date(day, month, year):
+    """
+    Convert day/month/year to a date object.
+
+    Arguments:
+        day (string): day of the date
+        month (string): month of the date
+        year (string): year of the date
+
+    Returns:
+        _date: date object if it is valid, otherwise None
+    """
+    from datetime import date
+    try:
+        _date = date(int(year), int(month), int(day))
+    except (ValueError, TypeError) as e:  # pylint: disable=unused-variable
+        return None
+
+    return _date
+
+
+def get_dates_from_duration(duration, **kwargs):
+    """
+    Returns the to and from dates based on the duration option selected by user.
+
+    Arguments:
+        duration (string): duration option selected by user
+
+    Returns:
+        from_day (date): starting date
+        to_day (date): Ending date
+    """
+    to_day = now()
+
+    if duration == LAST_6_MONTHS:
+        from_day = to_day - relativedelta(months=6)
+    elif duration == LAST_30_DAYS:
+        from_day = to_day - relativedelta(days=30)
+    elif duration == LAST_1_YEAR:
+        from_day = to_day - relativedelta(years=1)
```

Comment:
Can we achieve all of this by using a dict?

Reply:
It won't be consistent if we use dict for 3 options and if/else for the fourth

Reply:
imo an `if-else` with a dict is still a cleaner approach. But it's completely your choice!